[docs] Add 3D reduction example to tensordot docs

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45697 [docs] Add 3D reduction example to tensordot docs**

**Summary**
This commit adds an example of a reduction over three dimensions with
`torch.tensordot`. It is unclear from existing docs whether `dims`
should be a list of pairs or a pair of lists.

**Test Plan**
Built the docs locally.

*Before*
<img width="864" alt="Captura de Pantalla 2020-10-01 a la(s) 1 35 46 p  m" src="https://user-images.githubusercontent.com/4392003/94866838-f0b17f80-03f4-11eb-8692-8f50fe3b9863.png">

*After*
<img width="831" alt="Captura de Pantalla 2020-10-05 a la(s) 12 06 28 p  m" src="https://user-images.githubusercontent.com/4392003/95121092-670af600-0703-11eb-959f-73c7797a76ee.png">


**Fixes**
This commit closes #22748.

Differential Revision: [D24118186](https://our.internmc.facebook.com/intern/diff/D24118186)

Commit:
[docs] Add 3D reduction example to tensordot docs

**Summary**
This commit adds an example of a reduction over three dimensions with
`torch.tensordot`. It is unclear from existing docs whether `dims`
should be a list of pairs or a pair of lists.

**Test Plan**
Built the docs locally.

**Fixes**
This commit closes #22748.

[ghstack-poisoned]

SplitInfinity commented:
I'm not sure why the result shows up in scientific notation but it's not consistent with the rest of the docs. I'd like to adjust that if possible.

zou3519 commented on torch/functional.py:
```diff
@@ -919,6 +919,12 @@ def tensordot(a, b, dims=2):
+        >>> a = torch.randn(3, 5, 4, 6)
+        >>> b = torch.randn(6, 4, 5, 3)
+        >>> torch.tensordot(a, b, dims=([2, 1, 3], [1, 2, 0]))
+        tensor([[ 9.4136e+00, -1.0321e+01, -3.2591e+00],
+                [-9.8923e+00, -1.5904e+01,  7.2188e+00],
+                [ 5.1670e-03,  8.7355e+00,  1.9348e+01]])
```
You can adjust https://pytorch.org/docs/stable/generated/torch.set_printoptions.html#torch.set_printoptions before printing.

zou3519 commented on torch/functional.py:
```diff
@@ -884,7 +884,7 @@ def tensordot(a, b, dims=2):
       a (Tensor): Left tensor to contract
       b (Tensor): Right tensor to contract
-      dims (int or tuple of two lists of integers): number of dimensions to
+      dims (int or tuple of two lists of int): number of dimensions to
```
nit: I think `int or Tuple[List[Int]]` makes it a clearer (all of the "of"s throw me off a little)

SplitInfinity replied:
Fixed, but I used `Tuple[List[int]] of two lists` to indicate that the tuple must be a pair.

zou3519 replied:
Sorry, I didn't read that part. `Tuple[List[int]] of two lists` sounds good to me (y)

Commit:
Update on "[docs] Add 3D reduction example to tensordot docs"


**Summary**
This commit adds an example of a reduction over three dimensions with
`torch.tensordot`. It is unclear from existing docs whether `dims`
should be a list of pairs or a pair of lists.

**Test Plan**
Built the docs locally.

*Before*
<img width="864" alt="Captura de Pantalla 2020-10-01 a la(s) 1 35 46 p  m" src="https://user-images.githubusercontent.com/4392003/94866838-f0b17f80-03f4-11eb-8692-8f50fe3b9863.png">

*After*
<img width="897" alt="Captura de Pantalla 2020-10-01 a la(s) 2 37 30 p  m" src="https://user-images.githubusercontent.com/4392003/94866857-f4450680-03f4-11eb-98a2-34fbed2cf775.png">

**Fixes**
This commit closes #22748.

[ghstack-poisoned]
